be/src/exec/orc_scanner.cpp

```diff
@@ -0,0 +1,261 @@
+            RuntimeProfile* profile,
+            const TBrokerScanRangeParams& params,
+            const std::vector<TBrokerRangeDesc>& ranges,
+            const std::vector<TNetworkAddress>& broker_addresses,
+            ScannerCounter* counter) : BaseScanner(state, profile, params, counter),
+            _ranges(ranges),
+            _broker_addresses(broker_addresses),
+            // _splittable(params.splittable),
+            _next_range(0),
+            _cur_file_eof(true),
+            _scanner_eof(false) {}
+
+ORCScanner::~ORCScanner() {
+    close();
+}
+
+Status ORCScanner::open() {
+    return BaseScanner::open();
+}
+
+Status ORCScanner::get_next(Tuple* tuple, MemPool* tuple_pool, bool* eof) {
+    try {
+        SCOPED_TIMER(_read_timer);
+        // Get one line
+        while (!_scanner_eof) {
+            if (_cur_file_eof) {
+                RETURN_IF_ERROR(open_next_reader());
+                if (_scanner_eof) {
+                    *eof = true;
+                    return Status::OK();
+                } else {
+                    _cur_file_eof = false;
+                }
+            }
+            if (_current_line_of_group >= _rows_of_group) { // read next stripe
+                if (_current_group >= _total_groups) {
+                    _cur_file_eof = true;
+                    continue;
+                }
+                _rows_of_group = _reader->getStripe(_current_group)->getNumberOfRows();
+                std::cout << "rows of group "<< _rows_of_group << std::endl;
+                _batch = _row_reader->createRowBatch(_rows_of_group);
+                _row_reader->next(*_batch.get());
+
+                _current_line_of_group = 0;
+                ++_current_group;
+            }
+
+            std::vector<orc::ColumnVectorBatch *> _batch_vec = ((orc::StructVectorBatch *) _batch.get())->fields;
+            for (int column_ipos = 0; column_ipos < _num_of_columns_from_file; ++column_ipos) {
+                auto slot_desc = _src_slot_descs[column_ipos];
+                orc::ColumnVectorBatch *cvb = _batch_vec[_column_name_map_orc_index.find(
+                        slot_desc->col_name())->second];
+
+                if (cvb->hasNulls && !cvb->notNull[_current_line_of_group]) {
+                    if (!slot_desc->is_nullable()) {
+                        std::stringstream str_error;
+                        str_error << "The field name(" << slot_desc->col_name()
+                                  << ") is not allowed null, but ORC field is NULL.";
+                        LOG(WARNING) << str_error.str();
+                        return Status::RuntimeError(str_error.str());
+                    }
+                    _src_tuple->set_null(slot_desc->null_indicator_offset());
+                } else {
+                    int32_t wbytes = 0;
+                    uint8_t tmp_buf[128] = {0};
+                    _src_tuple->set_not_null(slot_desc->null_indicator_offset());
```

Comment:
should check if slot_desc is nullable

Reply:
all src tuple is nullable , so we judge null in dest_tuple and report error row